mongoose.c

```diff
@@ -1971,6 +1971,8 @@ struct threadparam {
 };
 
 static int wait_until_ready(sock_t sock, int for_read) {
+  if ( (sock < 0) || (sock >= FD_SETSIZE) )
```

Comment:
1. error build *.c file 
mongoose\mongoose.c(1989): error C2275: 'fd_set' : illegal use of this type as an expression

fix
static int wait_until_ready(sock_t sock, int for_read) {
  fd_set set;
  if ( (sock < 0) || (sock >= FD_SETSIZE) )
    return 0;
....